runtime/type-registry/src/lib.rs

```diff
@@ -1,16 +1,134 @@
-use std::collections::HashMap;
-use std::fmt;
-use std::fmt::Debug;
-use std::fmt::Display;
-use std::fmt::Formatter;
-use std::ptr;
-use std::sync::LazyLock;
-use std::sync::RwLock;
+#![no_std]
+/* use the C allocator; don't use libstd */
```

Comment:
Why do we need to avoid `std`?  It doesn't seem smart to instead write our own `Mutex`; there's a lot we can do incorrectly.

Reply:
If we're going to write our own `Mutex`, we should use `no_std` [`lock_api`](https://docs.rs/lock_api) to write it.